edge-util/src/Microsoft.Azure.Devices.Edge.Util/ConfigurationExtensions.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.Logging;
+
+    public static class ConfigurationExtensions
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger(typeof(ConfigurationExtensions));
+
+        public static T GetValueIgnoreException<T>(this IConfiguration configuration, string key, T defaultValue)
+        {
+            try
+            {
+                return configuration.GetValue(key, defaultValue);
+            }
+            catch (Exception ex)
```

Comment:
We don't want to ignore _all_ exceptions, just the ones that specifically come from malformed input. `TimeSpan.Parse` docs say it can throw `FormatException` or `OverflowException`; let's catch those (and add tests to validate). Since we're catching specific exceptions from `TimeSpan.Parse`, let's also scope this extension method down to just `TimeSpan` instead of `T`. Finally after all that, a more appropriate name might be `GetTimeSpanValue`?

Reply:
Changed to be timespan specific. It will be hard to unit test, since GetValue is an extension function and Moq can't mock it. It's a simple enough function I think its safe leaving it untested.

Reply:
Have a look at all the `edge-util/test/Microsoft.Azure.Devices.Edge.Util.Test/*ExTest.cs` files...it's easy to get some basic unit testing in place for the extension class, no mocks needed.

Reply:
Started to worry that it might be hard to test (due to Logger?), so tried it really quick. It worked an now I have some tests on my machine, so I pushed them. ;)